tests/unit/lib/SGVector_unittest.cc

```diff
@@ -9,6 +9,22 @@
 
 using namespace shogun;
 
+TEST(SGVectorTest, vectorized_rng)
+{
+	auto rng_check_float = [](auto elem) {
```

Comment:
This is a bit of a strange construct. You are passing a function argument to actually only used its type information (which in turn requires you to use the decltype). This is asking for a template.

Can you use a template lambda?

Otherwise, I'd prefer an auxiliary function outside the TEST body.

Reply:
IIRC templates lambdas arent supported till C++20. Here Im using concepts with C++14 itself. I'll put an auxiliary function outside sure.